debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -490,6 +504,9 @@ protected void handleInsert(Event event) throws InterruptedException {
             logger.debug("Skipping previously processed row event: {}", event);
             return;
         }
+        if (ignoreDmlEventByGtidSource) {
+            return;
```

Comment:
Probably should have a DEBUG log statement (similar to line 504) saying why it was skipped.